blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/EnumMetricDataIO.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (c) 2016 Rackspace.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.rackspacecloud.blueflood.io;
+
+import com.google.common.collect.Table;
+import com.rackspacecloud.blueflood.cache.MetadataCache;
+import com.rackspacecloud.blueflood.concurrent.ThreadPoolBuilder;
+import com.rackspacecloud.blueflood.exceptions.CacheException;
+import com.rackspacecloud.blueflood.outputs.formats.MetricData;
+import com.rackspacecloud.blueflood.rollup.Granularity;
+import com.rackspacecloud.blueflood.service.Configuration;
+import com.rackspacecloud.blueflood.service.CoreConfig;
+import com.rackspacecloud.blueflood.types.*;
+import com.rackspacecloud.blueflood.utils.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+
+/**
+ * This class uses the driver specific {@link EnumReader} to
+ * read/write enum metrics from Cassandra and convert them to
+ * {@link com.rackspacecloud.blueflood.outputs.formats.MetricData}
+ * objects that are used by services such as Rollup, Ingest and Query
+ */
+public class EnumMetricDataIO {
```

Comment:
I was under the impression that all *IO classes were direct implementations of of a particular Cassandra driver.  I think that's a pretty good thing to follow, albeit no better names are really coming to mind.

Some ideas:
* EnumMetrics
* EnumMetricData
* EnumMetricAccessor